mesonpy/__init__.py

```diff
@@ -351,7 +356,17 @@ def tag(self) -> mesonpy._tags.Tag:
     @property
     def name(self) -> str:
         """Wheel name, this includes the basename and tag."""
-        return f'{self._metadata.distribution_name}-{self._metadata.version}-{self.tag}'
+        name = f'{self._metadata.distribution_name}-{self._metadata.version}-{self.tag}'
+        if self._variant is not None:
+            name += f'-{self._variant.hexdigest}'
+            labels = PluginLoader().get_variant_labels(self._variant)
+            for numlabels in range(len(labels), 0, -1):
+                long_name = "+".join((name, *labels[:numlabels]))
+                # if labels would give us filename that's longer than 128
+                # characters (124 + .whl), strip them
```

Comment:
Does this take into account that we're running this code _before_ `auditwheel` is going to make it quite a bit longer?

Reply:
Makes sense. We can always tweak the number later. My gut feel is that 128 is on the generous side, but it doesn't matter too much for now.